### PR TITLE
Feature public recipes list

### LINKED ISSUE #6: [3pts] Public recipe list
- [ ] Should display a list of all public recipes ordered by the newest as in the wireframe.
- [ ] Should lead to recipe details.

### LINKED ISSUE #13: [2pts] N+1 queries happening
Fix N+1 queries happening, to identify and optimize N+1 queries occurring in a web application to improve performance and reduce unnecessary database load.

### PR DESCRIPTION
**Pull request by Atifzada**
#### Following changes have been made to this PR:

- [x] Displayed a list of all public recipes
- [x] Added public recipes page
- [x] Fixed linter errors

### CODE REVIEW

File: app/controllers/recipes_controller.rb [hunk 1]
```diff
@@ -3,18 +3,31 @@ class RecipesController < ApplicationController
...
+  #     # Display the recipe details as in the wireframe
+  #   else
+  #     redirect_to recipes_path, alert: 'You are not authorized to view this private recipe.'
+  #   end
+  # end
+
   def show
-    @recipe = Recipe.includes(:user, :recipe_foods).find(params[:id])
-    @recipe_food = RecipeFood.new
+    @recipe = Recipe.find(params[:id])
```
Comment (adeola003) [inline app/controllers/recipes_controller.rb]: The N+1 problem in the method was already fixed. kindly replace this line with the previous one `@recipe = Recipe.includes(:user, :recipe_foods).find(params[:id])`

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -17,6 +17,8 @@
         <li><%= link_to "Home", root_path %></li>
         <li><%= link_to "Recipes", recipes_path %></li>
         <li><%= link_to "New Recipe", new_recipe_path %></li>
+        <li><%= link_to "Foods", foods_path %></li>
+        <li><%= link_to "New Food", new_food_path %></li>
```
Comment (adeola003) [inline app/views/layouts/application.html.erb:21]: Nice job adding those elements to the menu bar. Consider adding as well a link to the public recipes and the shopping list.

**Review by adeola003 — CHANGES_REQUESTED** [review]
Nice Job so far. You did great implementing some project requirements. However, after implementing the requested changes, we will be one step away from completing our project.

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (Atifzada) [reply app/views/layouts/application.html.erb:21]: Thank you for your suggestions. Surely I will add it but still, there is some work left to implement completely Public recipes and shopping list that's why I avoided adding them at this stage

**Review by adeola003 — APPROVED** [review]
You did an excellent Job implementing the changes! Now let's move to the following requirements. You can merge this branch

**Pull request merged**